### PR TITLE
Fixes crash if cannot find --solver-path

### LINKED ISSUE #3572: Crash if cannot find given solver-path
### Dafny version

3.11.0+dev

### Code to produce this issue

```dafny
dafny verify --solver-plugin x test.dfy
```

### Command to run and resulting output

```code
File name: '/Users/davidcok/projects/dafny2/dafny/Binaries/Boogie.Provers.x.dll'
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at Microsoft.Boogie.ProverFactory.Load(String proverName)
   at Microsoft.Dafny.CommonOptionBag.<>c.<.cctor>b__29_23(DafnyOptions o, String v) in /Users/davidcok/projects/dafny2/dafny/Source/DafnyCore/Options/CommonOptionBag.cs:line 243
   at Microsoft.Dafny.DafnyOptions.<>c__DisplayClass8_0`1.<RegisterLegacyBinding>b__0(DafnyOptions options, Object o) in /Users/davidcok/projects/dafny2/dafny/Source/DafnyCore/DafnyOptions.cs:line 134
   at Microsoft.Dafny.DafnyOptions.ApplyBinding(Option option) in /Users/davidcok/projects/dafny2/dafny/Source/DafnyCore/DafnyOptions.cs:line 110
   at Microsoft.Dafny.CommandRegistry.<>c__DisplayClass6_0.<Create>g__CommandHandler|2(InvocationContext context) in /Users/davidcok/projects/dafny2/dafny/Source/DafnyDriver/Commands/CommandRegistry.cs:line 126
   at System.CommandLine.Invocation.AnonymousCommandHandler.Invoke(InvocationContext context)
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

### What happened?

Crashed if file is non-existent, may be also if empty string, or otherwise not a plugin

### What type of operating system are you experiencing the problem on?

Mac

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #3572

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyDriver/Commands/CommandRegistry.cs [hunk 1]
```diff
@@ -188,7 +188,14 @@ class WritersConsole : IConsole {
         }
 
         options.OptionArguments[option] = value;
-        dafnyOptions.ApplyBinding(option);
+        try {
+          dafnyOptions.ApplyBinding(option);
+        } catch (Exception e) {
+          failedToProcessFile = true;
```
Comment (keyboardDrummer) [inline Source/DafnyDriver/Commands/CommandRegistry.cs:194]: Could you rename `failedToProcessFile` to `errorOccurred` ?

**Review by keyboardDrummer — COMMENTED** [review]
I like that this PR adds a try-catch because it potentially catches many bugs, but I would have liked to also prevent an exception from being thrown in the first place.

Something like 

```csharp
    SolverPath.AddValidator(r => {
      var fi = r.GetValueOrDefault<FileInfo>();
      if (fi is { Exists: false }) {
        r.ErrorMessage = $"File {fi} does not exist";
      }
    });
```

could work.

File: Source/DafnyDriver/Commands/CommandRegistry.cs (hunk 1 above)
Comment (davidcok) [reply Source/DafnyDriver/Commands/CommandRegistry.cs]: OK ( though the name was chosen before this PR)

**Comment by davidcok** [comment]
> I like that this PR adds a try-catch because it potentially catches many bugs, but I would have liked to also prevent an exception from being thrown in the first place.
> 
> Something like
> 
> ```cs
>     SolverPath.AddValidator(r => {
>       var fi = r.GetValueOrDefault<FileInfo>();
>       if (fi is { Exists: false }) {
>         r.ErrorMessage = $"File {fi} does not exist";
>       }
>     });
> ```
> 
> could work.

Done. One drawback is that using ErrorMessage also means that the entire help message is printed out with the error -- the user has to scroll back to see the real error. My choice for errors about a specific (legal) option is not to write out the whole help by default.

**Pull request merged**